### PR TITLE
NN-867, NN-838 Activity / Work Shop Searching, Display and print

### PR DESCRIPTION
**Pull request by steverendell**
Functionally complete I think but tests to follow

### CODE REVIEW

File: backend/controllers/activityList.js [hunk 1]
```diff
@@ -1,26 +1,42 @@
...
   const viewModel = await getActivityList(req, res);
   res.json(viewModel);
 }));
 
-const getActivityList = (async (req, res) => {
-  req.query.usage = 'PROD';
-  const activities = await elite2Api.getActivityList(req, res).data;
+const getActivityList = async (req, res) => {
+  // once request is not used passed to downstream services we wont have to manipulate it in this dodgy way
+  switchDateFormat(req);
```
Comment (steverendell) [inline backend/controllers/activityList.js]: needs tidying

File: src/ResultsActivity/ResultsActivity.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+import moment from 'moment';
+import { Link } from "react-router-dom";
+
+class ResultsActivity extends Component {
+  displayBack () {
+    return (<div className="padding-top no-print"><Link id={`back_to_menu_link`} title="Back to selection screen link" className="link backlink" to="/whereaboutssearch" >
+      <img className="back-triangle" src="/images/BackTriangle.png" alt="" width="6" height="10"/> Back to selection screen</Link></div>);
+  }
+
+  getActivityName () {
```
Comment (steverendell) [inline src/ResultsActivity/ResultsActivity.js]: a.locationId is an integer, is there a better way to compare these?

File: backend/controllers/activityList.js (hunk 1 above)
Comment (JamieDraperUK) [reply backend/controllers/activityList.js]: Why not add start splitting request parameters out now? elite2api.getActivities(req,red,date) should be fairly straight forward

File: src/ResultsActivity/ResultsActivity.js (hunk 2 above)
Comment (JamieDraperUK) [reply src/ResultsActivity/ResultsActivity.js]: You could do this.props.activities.filter(a => a.locationId  === Number(this.props.activity)

File: src/ResultsActivity/ResultsActivity.js [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        this.props.handleSave(this.props.history);
+      }}>Save changes</button>
+      {this.isToday(this.props.date) &&
+      <button id="printButton" className="button greyButton rightHandSide" type="button" onClick={() => {
+        this.props.handlePrint();
+      }}><img className="print-icon" src="/images/Printer_icon.png" height="23" width="20"/> Print list</button>
+      }
+    </div>);
+
+    let activityTitle;
```
Comment (JamieDraperUK) [inline src/ResultsActivity/ResultsActivity.js]: Might not be a massive deal but this is duplication from the ResultsHouseBlock

**Pull request merged**